### PR TITLE
searh functionality

### PR DESCRIPTION
**Pull request by lewisrenfrew**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "@linn-it/linn-form-components-library",
-    "version": "10.6.12",
+    "version": "10.6.13",
```
Comment (atcampbell) [inline package.json]: Should this be 10.7.0 as it sa new feature?

File: src/components/SearchPanel.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    const menuEntries = menu
+        .map(s => s.columns)
+        .flat()
+        .map(c => c.categories)
+        .flat()
+        .map(i => i.items)
+        .flat();
+
+    const uniqueEntries = Object.values(
+        menuEntries.reduce((c, e) => {
```
Comment (atcampbell) [inline src/components/SearchPanel.js]: Could these have clearer var names just to make it a bit more readable?

File: src/components/SearchPanel.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+                        value={searchTerm}
+                        onChange={handleFieldChange}
+                        textFieldProps={{
+                            autoFocus: true
+                        }}
+                        placeholder="start typing..."
+                    />
+                    {searchTerm &&
+                        uniqueEntries
+                            .filter(e => e.title.toLowerCase().startsWith(searchTerm.toLowerCase()))
```
Comment (atcampbell) [inline src/components/SearchPanel.js]: Could there be a way  to search within the words as well? So searcing for 'accounts' would bring back sales accounts etc

Comment (lewisrenfrew) [reply src/components/SearchPanel.js]: Probably... filtering a big list on a .`includes()` is a lot more expensive than `.startsWith()` so I went with the latter. I could look into a debounce but I really like instantaneous feel of the search in its current form... Maybe `useMemo()` can help.

**Pull request merged**